.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,262 @@
+          git config --global advice.detachedHead false
+
+          # Will be used before as a cache key
+          export CPUHASH="$(cat /proc/cpuinfo | grep "model name" | head -n 1 | cut -d':' -f2 | md5sum)"
+      - name: Git checkout
+        uses: actions/checkout@v2-beta
+        with:
+          fetch-depth: 1
+      - name: Setup APT repositories
+        run: |
+          # We don't need this at all, and our
+          # builds often fails because Microsoft
+          # servers are unstable or even offline.
+          sudo rm -f /etc/apt/sources.list.d/dotnetdev.list
+          sudo rm -f /etc/apt/sources.list.d/azure*.list
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update --quiet --yes 1>/dev/null
+          sudo apt-get install --no-install-recommends -q -y re2c
+
+      - name: Get Zephir Parser Cache Key
+        id: zephir-parser-cache-key
+        run: echo "::set-output name=key::$(echo -n ${ZEPHIR_PARSER_VERSION}_${CPUHASH})"
+
+      - name: Cache Zephir Parser
+        uses: actions/cache@v1
+        with:
+          path: ~/php-zephir-parser
+          key: ${{ runner.os }}-${{ matrix.php }}-zephir-parser-${{ steps.zephir-parser-cache.outputs.key }}
+          restore-keys: ${{ runner.os }}-${{ matrix.php }}-zephir-parser-
+
+      - name: Install Zephir Parser
+        run: |
+          if [ ! -f ~/php-zephir-parser/LICENSE ]; then
+            rm -rf ~/php-zephir-parser
+            git clone -b "$ZEPHIR_PARSER_VERSION" \
+              --depth 1 \
+              -q https://github.com/phalcon/php-zephir-parser \
+              ~/php-zephir-parser
+          fi
+
+          cd ~/php-zephir-parser
+          phpize
+          ./configure --with-php-config=/usr/bin/php-config --enable-zephir_parser
+          make -j"$(getconf _NPROCESSORS_ONLN)"
+          sudo make install
+          echo 'extension="zephir_parser.so"' | sudo tee "/etc/php/7.4/cli/conf.d/zephir_parser.ini"
+
+      - name: Setup GitHub Token
+        run: |
+          # To increase the GitHub rate limit we're use GitHub authentication
+          if [ -n "${{ secrets.GH_TOKEN }}" ]; then
+            composer config github-oauth.github.com "${{ secrets.GH_TOKEN }}"
+          fi
+
+      - name: Install Zephir
+        run: |
+          wget --no-clobber -O "$HOME/zephir" "https://github.com/phalcon/zephir/releases/download/${ZEPHIR_VERSION}/zephir.phar"
+          chmod +x "$HOME/zephir"
+      - name: Generate C code
+        run: |
+          $HOME/zephir fullclean
+          $HOME/zephir generate
+          cd build
+          php gen-build.php
+          cd ..
+
+      - name: Create pecl package
+        id: pecl_create
+        run: |
+          cp build/php7/safe/config.w32 config.w32
+          cp build/php7/safe/phalcon.zep.c phalcon.zep.c
+          cp build/php7/safe/config.m4 config.m4
+          cp build/php7/safe/php_phalcon.h php_phalcon.h
+          cp build/php7/safe/phalcon.zep.h phalcon.zep.h
```

Comment:
just FYI, `safe` is TS version, thus we don't provide NTS one

Reply:
@sergeyklay Hmm. This exact config is use for Pecl as well. We have NTS and TS there https://pecl.php.net/package/phalcon/4.0.2/windows

Reply:
Could you please add selftest as a first step, eg:
```
php -v
```

Just to make sure what exactly we use